packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts

```diff
@@ -92,6 +102,8 @@ export class AzureCommunicationChatAdapter implements ChatAdapter {
   private chatClient: StatefulChatClient;
   private chatThreadClient: ChatThreadClient;
   private context: ChatContext;
+  /* @conditional-compile-remove-from(stable): FILE_SHARING */
+  private fileSharingAdapter: FileSharingAdapter;
```

Comment:
This should store a `FileUploadAdapter` (the public interface) not the `FileSharingAdapter` so that it can only use the public API. **If we expose `FileUploadAdapter` as public API.

And that is a good question: Do we want to expose `FileUploadAdapter` as a public API at all? There are already new methods in `ChatAdapter` that Contoso can override. `FileUploadAdapter` can remain internal implementation for us, until we feel the need to export that separately. If we do export it,

* We will need to provide a way for dependency injection so Contoso can provide a different implementation of `FileUploadAdpater` and also a way to construct the `FileSharingAdapter`
* Also, the naming can be a bit confusing here. We should rename `FileSharingAdapter` to `AzureCommunicationFileUploadAdapter`, to follow the established pattern.


I think we should unexport `FileUploadAdapter` right now. At that point, we don't a separate interface at all (Just use `FileSharingAdapter` (renamed if needed) here).

Reply:
We can't unexport `FileUploadAdapter` as it is being extended by `ChatAdapter`
I think it's ok to not prefix these plugin adapters with `AzureCommunication` as it helps identify that `AzureCommunicationChatAdapter` is the main adapter and these are plugins(Extensions)

Reply:
Ah, we're extending it. OK, that makes sense then, from an interface API perspective.

I was still thinking composition.

So `AzureCommunicationChatAdapter` implements `FileUploadAdapter` (due to `extends`) and there is no need for dependency injection.

And `FileSharingAdapter` need not implement `FileUploadAdapter` at all (it's just an internal implementation -- encapsulating the logic for file upload).

Reply:
Right, but it's good for ensuring there is no drift b/w FileSharingAdapter and FileUploadAdapter going forward.

Reply:
Any new functions added to the FileUploadAdapter need to be implemented in FileSharingAdapter.